cdap-runtime-ext-dataproc/src/test/java/io/cdap/cdap/runtime/spi/common/DataprocUtilsTest.java

```diff
@@ -55,5 +55,19 @@ public void testParseLabelsWithoutVal() {
     expected.put("k1", "");
     expected.put("k2", "");
     Assert.assertEquals(expected, DataprocUtils.parseKeyValueConfig("k1,k2=", ",", "="));
+    Map<String, String> got = DataprocUtils.parseKeyValueConfig("", ",", "=");
+  }
+
+  @Test
+  public void testParseEmptyLabels() {
+    Assert.assertEquals(new HashMap<>(),
+        DataprocUtils.parseKeyValueConfig("", ",", "="));
+  }
+
+  @Test
+  public void testParseEmptyKey() {
+    Map<String, String> expected = new HashMap<>();
+    expected.put("", "v1");
```

Comment:
I thought it's not expected to have an empty string as key?

Reply:
ok I see from the implementation that this would be expected, but that doesn't seem like correct behavior. We should never allow an empty string as a label key as that is invalid.

Reply:
Made the change to throw an IllegalArgumentException when the key is empty. Updated the test expectations also.

Reply:
Change reverted.